components/Repo/Add.js

```diff
@@ -55,44 +82,144 @@ const styles = {
+        )
+    )
+  }, [data, templateFilter])
+
+  return isTemplate ? (
+    <Dropdown
+      label='Vorlage'
+      items={schemaOptions}
+      value={schema}
+      onChange={item => {
+        onChange({
+          schema: item.value,
+          templateRepoId: undefined,
+          templatePrefix: undefined
+        })
+      }}
+    />
+  ) : (
+    <Loader
+      error={data.error}
+      loading={data.loading}
+      render={() => (
+        <Autocomplete
+          label='Vorlage'
+          value={template}
+          filter={templateFilter}
+          items={templateOptions}
+          onChange={newTemplate => {
+            setTemplate(newTemplate)
+            setTemplateFilter('')
+            onChange({
+              schema: newTemplate.value,
+              templateRepoId: newTemplate.repoId,
+              templatePrefix: newTemplate.slug + '-'
+            })
+          }}
+          onFilterChange={newFilter => {
+            if (!template || template.text !== newFilter) {
+              setTemplateFilter(newFilter)
+            }
+          }}
+          icon={<SearchIcon size={30} style={{ color: colors.lightText }} />}
+        />
+      )}
+    />
+  )
+})
+
 class RepoAdd extends Component {
   constructor(props) {
     super(props)
+    const schema = schemaKeys.includes('article') ? 'article' : schemaKeys[0]
     this.state = {
-      title: '',
-      template: templateKeys.includes('article') ? 'article' : templateKeys[0]
+      schema,
+      title: ''
     }
   }
-  getSlug(title) {
-    const { template } = this.state
-    const schema = schemas[template]
-    const prefix = [REPO_PREFIX, schema && schema.repoPrefix]
-      .filter(Boolean)
-      .join('')
-    const slug = [prefix, slugify(title)].join('')
-
-    return slug
+  getSlug() {
+    const { title, schema, templatePrefix } = this.state
+    const { isTemplate } = this.props
+    const prefix = isTemplate
+      ? [REPO_PREFIX]
+      : [REPO_PREFIX, templatePrefix || schemas[schema]?.repoPrefix]
+          .filter(Boolean)
+          .join('')
+    return [prefix, slugify(title)].join('')
   }
-  onSubmit(event) {
-    event.preventDefault()
-
-    const { title, template, error } = this.state
-    const slug = this.getSlug(title)
-    if (error || !title || slug.length > 100) {
-      this.handleTitle(title, true)
-      return
-    }
 
+  goToEdit({ slug }) {
+    const { title, schema, templateRepoId } = this.state
+    const { isTemplate } = this.props
     Router.replaceRoute('repo/edit', {
       repoId: [GITHUB_ORG, slug],
       commitId: 'new',
       title,
-      template
+      schema,
+      templateRepoId,
+      isTemplate
     }).then(() => {
       window.scrollTo(0, 0)
     })
   }
-  handleTitle(value, shouldValidate) {
+
+  onSubmit(event) {
+    event.preventDefault()
+
+    const { title, error } = this.state
+    const { isTemplate } = this.props
+    const slug = this.getSlug()
+    const maxSlugLength = isTemplate ? 30 : 100
```

Comment:
I was thinking of adding that slug limit in the `MetaData` form in `editor/modules/meta/ui`. If we do it here I'd recommend moving the `maxSlugLength` logic to `handleTitle` so it's also the same limit while typing.